search/intermediate_result.cpp

```diff
@@ -282,19 +283,42 @@ void FillDetails(FeatureType & ft, Result::Details & details)
 
   feature::TypesHolder const typesHolder(ft);
 
-  details.m_isHotel = ftypes::IsHotelChecker::Instance()(typesHolder);
-  if (details.m_isHotel && strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), details.m_stars))
-    details.m_stars = std::min(details.m_stars, osm::MapObject::kMaxStarsCount);
-  else
-    details.m_stars = 0;
+  uint8_t starsCount = 0;
+  bool isHotel = ftypes::IsHotelChecker::Instance()(typesHolder);
+  if (isHotel && strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), starsCount))
+    starsCount = std::min(starsCount, osm::MapObject::kMaxStarsCount);
 
   auto const cuisines = feature::GetLocalizedCuisines(typesHolder);
-  details.m_cuisine = strings::JoinStrings(cuisines, osm::MapObject::kFieldsSeparator);
+  std::string cuisine = strings::JoinStrings(cuisines, osm::MapObject::kFieldsSeparator);
 
   auto const roadShields = ftypes::GetRoadShieldsNames(ft);
-  details.m_roadShields = strings::JoinStrings(roadShields, osm::MapObject::kFieldsSeparator);
+  std::string roadShield = strings::JoinStrings(roadShields, osm::MapObject::kFieldsSeparator);
 
-  details.m_fee = feature::GetLocalizedFeeType(typesHolder);
+  std::string fee = feature::GetLocalizedFeeType(typesHolder);
+    
+  std::string description;
+  
+  if (isHotel && starsCount > 0)
+  {
+    std::string stars;
+    for (int i = 0; i < starsCount; ++i)
+      stars.append(osm::MapObject::kStarSymbol);
+    description = stars;
+  }
```

Comment:
```suggestion
    description.assign(starsCount, osm::MapObject::kStarSymbol);
```

Reply:
This doesn't work. Assign does not repeat the star symbol

Reply:
Are you sure? https://godbolt.org/z/7rPWcaP17

Reply:
The problem is that `'★'` (which is different to `'*'`) doesn't seem to fit in a char, so it doesn't work properly. https://godbolt.org/z/cYfcKTT6h